python-package/lightgbm/basic.py

```diff
@@ -1691,6 +1662,9 @@ def __init__(self, params=None, train_set=None, model_file=None, model_str=None,
             if not isinstance(train_set, Dataset):
                 raise TypeError('Training data should be Dataset instance, met {}'
                                 .format(type(train_set).__name__))
+            train_set.construct()
```

Comment:
Shouldn't network initialization go before dataset construction?

Reply:
@guolinke  ping

Reply:
sorry, will fix it